### ISSUE TITLE
Testing timed observables - fakeasync support?

### ISSUE THREAD

**Issue by iangregsondev** [nestjs/nest#2395]
## Is your feature request related to a problem? Please describe.
In angular, there is the concept of fakeasync which allows us to test asyncronous code without waiting for a timeout, this also works great for observables that have intervals , delays etc

Providing fakeasync in nestjs would really be helpful. It appears that it depends on zonejs - 

Provide concerte examples (docs) for using fakeasync

## Describe the solution you'd like

Providing fakeasync in nestjs would really be helpful. It appears that it depends on zonejs - 

Provide concerte examples (docs) for using fakeasync

## What is the motivation / use case for changing the behavior?
Being able to use fakeasync so moving between angular and nestjs is transparent. Also the ability to test async code is a must in my honest opinion.

**Comment by kamilmysliwiec** [comment]
Could you elaborate on what exactly you want to achieve?

**Comment by appsolutegeek** [comment]
Sure, so if i have an observable to has an interval or delay and a timeout - all observable functions, I don't want to wait for the timeout to occur just to test that timeout did actually happen.

In angular we can do this using fakeasync, which allows us to control time. I was under the understanding that this function that is part of angular needs zone.js to function.

I must admit, i didn't test it - due to the fact that zonejs is involved and its an angular function.

more information is available here https://www.joshmorony.com/testing-asynchronous-code-with-fakeasync-in-angular/

and here https://blog.angularindepth.com/rxjs-testing-with-fake-time-94114271eed2

So to summarize, something to allow us to test for Timeouts (operator on rxjs) without actually waiting for the timeout to occur 

Does that make sense ?

Making our tests fast even though there are delays, intervals and timeouts involved.

**Comment by kamilmysliwiec** [comment]
This should be implemented at the testing framework level. Both Jest and Jasmine provide an API to create fake timers.
